src/suites/cts/gpu_test.ts

```diff
@@ -153,4 +191,48 @@ export class GPUTest extends Fixture {
     }
     return undefined;
   }
+
+  checkBufferWithRowPitch(
+    actual: Uint8Array,
+    exp: Uint8Array,
+    width: number,
+    height: number,
+    rowPitch: number,
+    bytesPerPixel: number
+  ): string | undefined {
```

Comment:
same here, can we deduplicate some of this code with checkBuffer?

Reply:
Done.